### ISSUE TITLE
May need to manually call ldconfig after make install

### ISSUE THREAD

**Issue by guillerodriguez** [pagekite/libpagekite#40]
On fresh installs of Ubuntu 14.04 and Ubuntu 16.04, building and installing libpagekite does not update the ld cache.

```
$ pagekitec
pagekitec: error while loading shared libraries: libpagekite.so.1: cannot open shared object file: No such file or directory
$ ls /usr/local/lib/libpagekite.so*
/usr/local/lib/libpagekite.so    /usr/local/lib/libpagekite.so.1.0.0
/usr/local/lib/libpagekite.so.1
$
```

Manually running `ldconfig` after `make install` fixes the problem.

This is apparently due to the Makefile calling `libtool -n` which builds symlinks as needed but skips updating the cache.

References:
 * https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=684981
 * https://github.com/esnet/iperf/issues/153
 * https://github.com/AravisProject/aravis/issues/26

I don't know if it is reasonable to try to fix this but perhaps it would be a good idea to document it (for example print a message from `make install`)

**Comment by BjarniRunar** [comment]
Good catch. I wonder, is there any harm in calling ldconfig for the user? Is that ever a *bad* thing?

**Comment by guillerodriguez** [comment]
That (always calling `ldconfig`) is suggested here: https://github.com/esnet/iperf/issues/153#issuecomment-42897883

Always calling ldconfig has two problems:

 1. Installing as non-root user (e.g. installing to a private dir). If invoked as a non-root user, `ldconfig` will complain because it does not have sufficient permissions. See: https://github.com/esnet/iperf/issues/153#issuecomment-45652503

 2. This woud fail on Mac OS X where `ldconfig` is not used. See: https://github.com/esnet/iperf/issues/153#issuecomment-45668565

Having said that: I guess you could always try to call `ldconfig` with stderr sent to `/dev/null`. If the user is root and we're not on Mac OS X, it will work as intended. If the user is not root or we're on Mac OS X, nothing breaks either.
